spring-vault-core/src/main/java/org/springframework/vault/config/ClientHttpConnectorFactory.java

```diff
@@ -87,7 +88,7 @@ private static void configureSsl(SslConfiguration sslConfiguration,
 			}
 		}
 		catch (GeneralSecurityException | IOException e) {
-			throw new IllegalStateException(e);
+			throw new VaultClientException("While configuring ssl", e);
```

Comment:
Let's keep it as `VaultException`.

Reply:
If anything is more local than specifying ones local ssl config, what is?

Fundamentally let's see if we agree on this:

There are exceptions that are local in origin, and from remote access.
That's the raison d'etre.

Reply:
Agree there's a local/transport/remote distinction. If anything is not transport/remote-related, then it happens locally. No need to state the obvious to the user as long as we are consistent with the remote part.